lib/cloud-metadata/aws.js

```diff
@@ -118,50 +95,36 @@ function getMetadataAwsV2 (host, port, socketTimeoutMs, httpTimeout, protocol, c
         finalData.push(data)
       })
 
-      res.on('end', function (data) {
-        try {
-          const headers = {
-            'X-aws-ec2-metadata-token': finalData.join('')
-          }
-          getMetadataAws(
-            host,
-            port,
-            headers,
-            socketTimeoutMs,
-            httpTimeout,
-            protocol,
-            cb
-          )
-        } catch (e) {
-          cb(e)
+      res.on('end', function () {
+        const token = sanitizeHttpHeaderValue(finalData.join(''))
+        const headers = {}
+        if (token && res.statusCode === 200) {
+          // uses return value from call to latest/api/token as token,
+          // and takes extra steps to ensure characters are valid
+          headers['X-aws-ec2-metadata-token'] = token
         }
+
+        getMetadataAwsWithHeaders(
+          baseUrl,
+          headers,
+          connectTimeoutMs,
+          resTimeoutMs,
+          logger,
+          cb
+        )
       })
     }
   )
   req.on('timeout', function () {
-    req.destroy()
-    if (!requestHasErrored) {
-      const error = Error('request for metadata token timed out')
-      requestHasErrored = true
-      cb(error)
-    }
+    req.destroy(new Error('request for metadata token timed out'))
   })
 
   req.on('connectTimeout', function () {
-    req.destroy()
-    if (!requestHasErrored) {
-      requestHasErrored = true
-      const error = Error('socket connection to metadata token server timed out')
-      cb(error)
-    }
+    req.destroy(new Error('socket connection to metadata token server timed out'))
   })
 
   req.on('error', function (error) {
```

Comment:
style nit: s/error/err/

Reply:
@trentm Happy to make this change -- but what's the rule we're following here?  My previous understanding was `err` for for callbacks, but `error` was OK in other places.  Do we want `err` everywhere, or just callbacks and events? (also, dings the "let's think about custom linter rules for the future" bell)

Reply:
"nit:" so happy to leave it to separate discussion.

This *is* a callback, in this case. I'm an `err` or `someContextErr` naming kinda guy and that naming seems (at least to me in my silo) to be fairly common in the community.

Again, I agree that considering looking at automatic lint/style-checker handling sounds good.